core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -493,6 +490,9 @@ public boolean removeRule(RelOptRule rule) {
       return false;
     }
 
+    // Remove rule name.
+    ruleNames.remove(rule.toString());
+
```

Comment:
It's interesting that this is only called during planner clean up phase. But I think it's still good to fix this.